frontend/src/concepts/pipelines/apiHooks/mlmd/__tests__/useMlmdContext.spec.ts

```diff
@@ -0,0 +1,96 @@
+import { testHook, standardUseFetchState } from '~/__tests__/unit/testUtils/hooks';
+import { MetadataStoreServicePromiseClient, Context } from '~/third_party/mlmd';
+import { MlmdContextTypes } from '~/concepts/pipelines/apiHooks/mlmd/types';
+import { usePipelinesAPI } from '~/concepts/pipelines/context';
+import { useMlmdContext } from '~/concepts/pipelines/apiHooks/mlmd/useMlmdContext';
+import { GetContextByTypeAndNameResponse } from '~/third_party/mlmd/generated/ml_metadata/proto/metadata_store_service_pb';
+
+// Mock the usePipelinesAPI hook
+jest.mock('~/concepts/pipelines/context', () => ({
+  usePipelinesAPI: jest.fn(),
+}));
+
+// Mock the getMlmdContext function
+jest.mock('~/concepts/pipelines/apiHooks/mlmd/useMlmdContext', () => ({
+  ...jest.requireActual('~/concepts/pipelines/apiHooks/mlmd/useMlmdContext'),
+  getMlmdContext: jest.fn(),
+}));
+
+// Mock the MetadataStoreServicePromiseClient
+jest.mock('~/third_party/mlmd', () => {
+  const originalModule = jest.requireActual('~/third_party/mlmd');
+  return {
+    ...originalModule,
+    MetadataStoreServicePromiseClient: jest.fn().mockImplementation(() => ({
+      getContextByTypeAndName: jest.fn(),
+    })),
+    GetContextByTypeAndNameRequest: originalModule.GetContextByTypeAndNameRequest,
+  };
+});
+
+describe('useMlmdContext', () => {
+  const mockClient = new MetadataStoreServicePromiseClient('');
+  const mockUsePipelinesAPI = jest.mocked(
+    usePipelinesAPI as () => Partial<ReturnType<typeof usePipelinesAPI>>,
+  );
+  const mockGetContextByTypeAndName = jest.mocked(mockClient.getContextByTypeAndName);
+
+  const mockContext = new Context();
+  mockContext.setName('test-context');
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockUsePipelinesAPI.mockReturnValue({
+      metadataStoreServiceClient: mockClient,
+    });
+  });
+
+  it('should return an error when no type is provided', async () => {
+    const renderResult = testHook(useMlmdContext)('someName');
+
+    expect(renderResult.result.current).toStrictEqual(
+      standardUseFetchState(null, false, undefined),
+    );
+  });
+
+  it('should return an error when no name is provided', async () => {
+    const renderResult = testHook(useMlmdContext)(undefined, MlmdContextTypes.RUN);
+
+    expect(renderResult.result.current).toStrictEqual(
+      standardUseFetchState(null, false, undefined),
+    );
+  });
```

Comment:
Need to wait for update and then assert an error is returned.

Reply:
same here, those are not ready errors so it is caught in use fetch state